src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -290,6 +290,16 @@ public void Remove(object key)
             StartScanForExpiredItems();
         }
 
+        /// <inheritdoc />
+        public void Clear()
+        {
+            CheckDisposed();
+            if (!_entries.IsEmpty)
```

Comment:
There is a race condition between `IsEmpty` and `Clear`. I'm not sure if it will be dangerous, though.
Is it possible to just call `Clear` (i.e. without the check for empty first)?

Reply:
Ohh, yes we can call directly Clear(), but I added that check just to avoid unnecessary calls to Clear().

Reply:
Maybe it would be better if the concurrent dicitonary (CD) handles this case. I mean if `Clear` is called and the CD is empty, it just nops. Just as thought which is not finished...

Reply:
I checked the implementation of Clear() method, I did not see any check there, everytime it's initializing new table under the hood with given size! So I think it's better to have this check here.
But am still open for any changes if required!

Reply:
This is been fixed with my other PR - https://github.com/dotnet/runtime/pull/44581